packages/js/src/socket/socket.ts

```diff
@@ -0,0 +1,99 @@
+import io, { Socket as SocketIO } from 'socket.io-client';
+import { BaseModule } from '../base-module';
+
+import {
+  NotificationReceivedEvent,
+  NotificationUnseenEvent,
+  NotificationUnreadEvent,
+  NovuEventEmitter,
+  SocketEventNames,
+} from '../event-emitter';
+import { Notification } from '../feeds';
+import { Session, TODO, WebSocketEvent } from '../types';
+
+const PRODUCTION_SOCKET_URL = 'https://ws.novu.co';
+const NOTIFICATION_RECEIVED: NotificationReceivedEvent = 'notifications.notification_received';
+const UNSEEN_COUNT_CHANGED: NotificationUnseenEvent = 'notifications.unseen_count_changed';
+const UNREAD_COUNT_CHANGED: NotificationUnreadEvent = 'notifications.unread_count_changed';
+
+export class Socket extends BaseModule {
+  #token: string;
+  #emitter: NovuEventEmitter;
+  #socketIo: SocketIO | undefined;
+  #socketUrl: string;
+
+  constructor({ socketUrl }: { socketUrl?: string }) {
+    super();
+    this.#emitter = NovuEventEmitter.getInstance();
+    this.#socketUrl = socketUrl ?? PRODUCTION_SOCKET_URL;
+  }
+
+  protected onSessionSuccess({ token }: Session): void {
+    this.#token = token;
+  }
+
+  #notificationReceived = ({ message }: { message: TODO }) => {
+    this.#emitter.emit(NOTIFICATION_RECEIVED, {
+      result: new Notification(message),
+    });
+  };
+
+  #unseenCountChanged = ({ unseenCount }: { unseenCount: number }) => {
+    this.#emitter.emit(UNSEEN_COUNT_CHANGED, {
+      result: unseenCount,
+    });
+  };
+
+  #unreadCountChanged = ({ unreadCount }: { unreadCount: number }) => {
+    this.#emitter.emit(UNREAD_COUNT_CHANGED, {
+      result: unreadCount,
+    });
+  };
```

Comment:
Handle different types of WebSocket events and just call our emitter.

Reply:
you can check the three above functions and their interfaces, that's what is called and returned